infra/main.tf

```diff
@@ -14,8 +14,13 @@
  * limitations under the License.
  */
 locals {
-  nextauth_url      = "http://${google_compute_global_address.default.address}"
-  firestore_enabled = length(data.google_cloud_asset_resources_search_all.firestore_database.results) == 1 ? true : false
+  nextauth_url                   = "http://${google_compute_global_address.default.address}"
+  has_default_firestore_database = length(data.google_cloud_asset_resources_search_all.default_firestore_database.results) > 0 ? true : false
+}
+
+# Make a GET request to check for Firestore databases named "(default)"
+data "google_cloud_asset_resources_search_all" "default_firestore_database" {
+  query = "name=\"projects/${var.project_id}/databases/(default)\""
 }
```

Comment:
The `google_cloud_asset_resources_search_all` is documented here: https://registry.terraform.io/providers/hashicorp/google/latest/docs/data-sources/cloud_asset_resources_search_all

It queries the Google Cloud Asset Inventory to find resources within the specified scope (project, folder, or organization).